### PR TITLE
UserFacade and adding userService method

### PR DESCRIPTION
**Pull request by katHermanova**

### CODE REVIEW

File: library-api/src/main/java/cz/muni/fi/pa165/library/dto/UserDTO.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * @author Katarína Hermanová
+ * UČO 433511
+ * Github katHermanova
+ */
+public class UserDTO {
+    private long id;
+    private String firstName;
+    private String lastName;
+    private String email;
+    private String password;
```
[the remark this answers is not in the document]
Comment (katHermanova) [reply library-api/src/main/java/cz/muni/fi/pa165/library/dto/UserDTO.java:13]: I will look at it, but I tried to make it as it was in school project.

File: library-api/src/main/java/cz/muni/fi/pa165/library/facade/UserFacade.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+    UserDTO findById(long id);
+    List<UserDTO> findByFirstName(String firstName);
+    List<UserDTO> findByLastName(String lastName);
+    List<UserDTO> findByEmail(String email);
+    List<UserDTO> findAll();
+    void addUser(UserDTO user);
+    void deleteUser(long id);
+    void deleteAllUsers();
+    long count();
```
[the remark this answers is not in the document]
Comment (katHermanova) [reply library-api/src/main/java/cz/muni/fi/pa165/library/facade/UserFacade.java:22]: if you mean method count(), it is the same method that is in the UserService...I will delete it from everywhere as well as deleteAllUsers() to make our lives not so complicated, but I can not delete CRUD method without penalisation on my side so I will not do that, I am sorry

File: library-service/src/main/java/cz/muni/fi/pa165/library/services/UserService.java [hunk 3]
```diff
@@ -174,4 +174,23 @@ public void deleteAllUsers() {
...
     }
+
+    /**
+     *
+     * @param user to authenticate
+     * @param password entered password
+     * @return if entered password is valid
+     * @throws IllegalArgumentException if user does not have any password associated with him
+     */
+    public boolean authenticate(User user, String password) {
```
[the remark this answers is not in the document]
Comment (katHermanova) [reply library-service/src/main/java/cz/muni/fi/pa165/library/services/UserService.java:193]: I will look at that. Thanks

**Comment by katHermanova** [comment]
I am sorry it is a duplicate, but if I understand correctly from the 2nd Milestone, it should be where I put it. "You must have at least 2 non-trivial business functions on Service Layer (the example project contains several of them)." and "Everything that should happen in the system must be available through these interfaces (create entities, deleting them etc);"..maybe I understand it wrongly and I am sorry if that is the case. Anyway, please be more specific with what you mean.

**Comment by petr7555** [comment]
> I am sorry it is a duplicate, but if I understand correctly from the 2nd Milestone, it should be where I put it. "You must have at least 2 non-trivial business functions on Service Layer (the example project contains several of them)." and "Everything that should happen in the system must be available through these interfaces (create entities, deleting them etc);"..maybe I understand it wrongly and I am sorry if that is the case. Anyway, please be more specific with what you mean.

Authentication is not a business function. Converting an image, calculating an interest etc. are business functions.

**Comment by katHermanova** [comment]
https://github.com/fi-muni/PA165/blob/master/eshop-service/src/main/java/cz/fi/muni/pa165/service/UserService.java .... I still need to have at least 2 more methods in service layer. I ask you again, as I did in facebook conversation, please tell me, what do you then want me to add so we both don't waste our time. There is till plenty of work to do.

**Comment by petr7555** [comment]
You can sketch out the ui, the interactions of the user and figure out, which methods will be needed. I used Figma. If you want, I can send you an edit link so you can stick with the same style. Also our use case diagram could be helpful.

**Comment by katHermanova** [comment]
Sure, send me the link. Thank you

**Pull request closed without merging**